scripts/prepare-machine.ps1

```diff
@@ -4,147 +4,167 @@
+    Write-Host "No arguments passed, defaulting -ForBuild and -ForTest"
+    $ForBuild = $true
+    $ForTest = $true
+    if ("" -eq $Tls -and !$ForKernel) { $Tls = "openssl" }
+}
 
-    if ($Tls -eq "openssl") {
-        Write-Host "Initializing openssl submodule"
-        git submodule init submodules/openssl
-        git submodule update
-    }
+if ($ForBuild) {
+    # When configured for building, make sure we have all possible dependencies
+    # enabled for any possible build.
+    $InstallNasm = $true
+    $InstallJom = $true
+    $InstallXdp = $true
+    $InitSubmodules = $true
+}
 
-    if ($Kernel) {
-        # Remove OpenSSL
-        git rm submodules/openssl
-    }
+if ($ForTest) {
+    # When configured for testing, make sure we have all possible dependencies
+    # enabled for any possible test.
+    $InstallSigningCertificate = $true
+    $InstallTestCertificates = $true
 
-    if (!$Extra.Contains("-DisableTest")) {
-        Write-Host "Initializing googletest submodule"
-        git submodule init submodules/googletest
-        git submodule update
-    }
+    #$InstallCodeCoverage = $true # Ideally we'd enable this by default, but it
+                                  # hangs sometimes, so we only want to install
+                                  # for jobs that absoultely need it.
 }
 
-Write-Host "Initializing clog submodule"
-git submodule init submodules/clog
-git submodule update
+# Default TLS based on current platform.
+if ("" -eq $Tls) {
+    if ($IsWindows) { $Tls = "schannel" }
+    else            { $Tls = "openssl" }
+}
 
+# Root directory of the project.
+$RootDir = Split-Path $PSScriptRoot -Parent
 $ArtifactsPath = Join-Path $RootDir "artifacts"
+if (!(Test-Path $ArtifactsPath)) { mkdir $ArtifactsPath }
+
+# Directory for the corenet CI install.
 $CoreNetCiPath = Join-Path $ArtifactsPath "corenet-ci-main"
 $SetupPath = Join-Path $CoreNetCiPath "vm-setup"
 
+# Generic, default password for test certificates.
+$PfxPassword = ConvertTo-SecureString -String "placeholder" -Force -AsPlainText
+
+# Downloads and caches the latest version of the corenet-ci-main repo.
 function Download-CoreNet-Deps {
     # Download and extract https://github.com/microsoft/corenet-ci.
-    if (!(Test-Path $ArtifactsPath)) { mkdir $ArtifactsPath }
+    if ($Force) { rm -Force -Recurse $CoreNetCiPath -ErrorAction Ignore }
     if (!(Test-Path $CoreNetCiPath)) {
+        Write-Host "Downloading CoreNet-CI"
         $ZipPath = Join-Path $ArtifactsPath "corenet-ci.zip"
         Invoke-WebRequest -Uri "https://github.com/microsoft/corenet-ci/archive/refs/heads/main.zip" -OutFile $ZipPath
         Expand-Archive -Path $ZipPath -DestinationPath $ArtifactsPath -Force
         Remove-Item -Path $ZipPath
     }
 }
 
-function Download-Xdp-Kit {
-    if ($Force) { rm -Force -Recurse $ArtifactsPath }
-    if (!(Test-Path $ArtifactsPath)) { mkdir $ArtifactsPath }
+# Downloads the latest version of XDP (for building).
+function Install-Xdp {
```

Comment:
But this isn't really installing XDP, it's downloading it. And we will be needing this script to actually INSTALL xdp for the xdp tests.

Reply:
It can be adapted to do that when you onboard XDP tests.

Reply:
I don't get it. This name is misleading. What was wrong with the original name?

Reply:
When you "install" an SDK, which consists of just headers and libraries used to build, is that not installing in your opinion?

Reply:
I agree that the term "install" is commonly used for SDKs when there is no potential confusion. But that isn't the case here. The XDP SDK can be installed, or the XDP driver can be installed.

It seems to me like your focus is on the name being consistent with other "Install-Xxx" functions. If so, then how about Install-Xdp-Sdk?

Reply:
Ok. I can do that.